packages/cssnano-preset-advanced/src/index.js

```diff
@@ -9,6 +9,9 @@ const defaultOpts = {
     autoprefixer: {
         add: false,
     },
+    cssDeclarationSorter: {
+        customOrder: null,
```

Comment:
Explain your reasoning behind having a different sorting order to the advanced preset and the consequences it'll have.

Reply:
The advanced order can make unsafe changes to save more bytes, so it orders everything.

Reply:
That's not really what the advanced preset is about, though. See the website on 'advanced transforms' for more info.

Reply:
From the advanced transform page: "Certain optimizations are not suitable for all use cases; unlike those that we bundle by default, advanced transforms all carry a certain amount of risk.". This exactly falls into this category. When using legacy code with double properties that override each other sorting everything comes with a certain amount of risk.

Reply:
The risk with the advanced preset comes from mixing sources. When working within the same stylesheet, it should be safe. The `advanced` preset is not "bug mode".

Reply:
I disagree, to be more specific also quoting the same page: "This transform changes CSS semantically; it may remove styles based on certain characteristics of the CSS, or it might update values to make the CSS smaller overall. Depending on exactly what the transform does, this may be undesirable because it might change the appearance of your website in certain use cases.". Is what this optimization falls under, I think reducing the size here further is worth the tiny risk of breaking legacy code which other optimizations in this preset also do.

Thoughts @ai @evilebottnawi ?

Reply:
>Certain optimizations are not suitable for all use cases [...] we've documented the assumptions that it must make about your CSS in order for the transform to be **safe**. [...] these transforms fall into either or both of categories.

>Assumes concatenation
> [...] assumes that the CSS passed through cssnano is all that is needed for a website to run [...]
>may be problematic if the styles you are writing in some way interact with third party styles,  or you are using multiple stylesheets instead of concatenating them 

>Changes semantics

>may remove styles based on certain characteristics of the CSS, or it might update values to make the CSS smaller overall. [...]  this *may* be undesirable [...]
>A good example is autoprefixer

What you are going is against the spirit of the preset. It has clear qualifiers on what it takes to use the advanced preset safely. It's not supposed to be a roll of the dice on whether it works or not. Please respect that.

Reply:
It's not a "roll of the dice" though, it is not safe when the CSS contains conflicting properties in one selector. I don't see how this is different from e.g. autoprefixer that removes outdated properties that is unsafe for CSS that contains outdated properties.

Reply:
> it is not safe when the CSS contains conflicting properties in one selector.

Yeah, we can't have that. It's a super-common thing in CSS to have properties that affect the same space. Bringing this into the advanced preset would destroy any use case it has, rendering it useless.

Reply:
So being not super-common it would be useful... Is it super-common? For one, this issue took a while to surface which I think partly indicates this is uncommon. Secondly as far as I know CSS selectors normally do not contain properties that override each other, it's usually a mistake, e.g:

````
animation: bla;
animation-name: bla;
```

The `animation` property also contains the name so instead of putting `animation-name` after it normally I would just change the name in the `animation` shorthand value.